girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -219,8 +219,9 @@ def finalizeUpload(self, upload, file):
 
         # Store the hash in the upload so that deleting a file won't delete
         # this file
-        upload['sha512'] = hash
-        self.model('upload').update({'_id': upload['_id']}, update={'$set': {'sha512': hash}})
+        if '_id' in upload:
```

Comment:
I've left the bulk of the code in place, but now it doesn't ever ask to not save uploads.  I can change the message about future changes to something that has a "for Girder 3" text string in it if we have such a thing.  Once this goes in, I'll create the 3.0/performance issue.